speech/google/cloud/speech/client.py

```diff
@@ -299,6 +298,74 @@ def sync_recognize(self, sample, language_code=None,
                                   profanity_filter, speech_context)
 
 
+class StreamingSpeechResult(object):
+    """Streaming speech result representation.
+
+    :type alternatives: list
+    :param alternatives: List of protobuf speech alternatives.
+
+    :type is_final: bool
+    :param is_final: Boolean indicator of results finality.
+
+    :type stability: float
+    :param stability: 0.0-1.0 stability score for the results returned.
+
+    :rtype: :class:`~google.cloud.speech.client.StreamingSpeechResult`
+    :returns: Instance of ``StreamingSpeechResult``.
+    """
+    def __init__(self, alternatives, is_final=False, stability=0.0):
+        self._alternatives = [Alternative.from_pb(alternative)
+                              for alternative in alternatives]
+        self._is_final = is_final
+        self._stability = stability
+
+    @classmethod
+    def from_pb(cls, response):
+        """Factory: construct instance of ``StreamingSpeechResult``.
+
+        :type response: :class:`~google.cloud.grpc.speech.v1beta1\
+                               .cloud_speech_pb2.StreamingRecognizeResult`
+        :param response: Istance of ``StreamingRecognizeResult`` protobuf.
+
+        :rtype: :class:`~google.cloud.speech.client.StreamingSpeechResult`
+        :returns: Instance of ``StreamingSpeechResult``.
+        """
+        alternatives = response.alternatives
+        is_final = response.is_final
+        stability = response.stability
+        return cls(alternatives=alternatives, is_final=is_final,
+                   stability=stability)
+
+    @property
+    def alternatives(self):
+        """List of alternative transcripts.
+
+        :rtype: list of :class:`~google.cloud.speech.alternative.Alternative`
+        :returns: List of ``Alternative`` instances.
+        """
+        return self._alternatives
+
+    @property
+    def is_final(self):
```

Comment:
This (making it read-only) is overkill. Just set `self.is_final` on the instance.